packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -757,11 +758,18 @@ async function handler(
     throw new Error(ErrorCode.NoAvailableUsersFound);
   }
 
+  type organizerWithUnMaskedEmail = (typeof users)[number] & { maskedEmail?: string };
+
   // If the team member is requested then they should be the organizer
-  const organizerUser = reqBody.teamMemberEmail
+  const organizerUser: organizerWithUnMaskedEmail = reqBody.teamMemberEmail
     ? users.find((user) => user.email === reqBody.teamMemberEmail) ?? users[0]
     : users[0];
 
+  const hideOrganizerEmail = eventType.team?.hideOrganizerEmail;
+  if (hideOrganizerEmail) {
+    organizerUser.maskedEmail = maskEmail(organizerUser);
+  }
+
```

Comment:
The idea is to add a new property `maskedEmail` which then will be used on booking success page and calendar description